src/librustdoc/passes/check_code_block_syntax.rs

```diff
@@ -16,85 +15,24 @@ pub const CHECK_CODE_BLOCK_SYNTAX: Pass = Pass {
     description: "validates syntax inside Rust code blocks",
 };
 
-pub fn check_code_block_syntax(krate: clean::Crate, cx: &DocContext<'_>) -> clean::Crate {
-    SyntaxChecker { cx }.fold_crate(krate)
+pub fn check_code_block_syntax(krate: clean::Crate, _cx: &DocContext<'_>) -> clean::Crate {
+    SyntaxChecker { _cx }.fold_crate(krate)
 }
 
 struct SyntaxChecker<'a, 'tcx> {
-    cx: &'a DocContext<'tcx>,
+    _cx: &'a DocContext<'tcx>,
 }
 
 impl<'a, 'tcx> SyntaxChecker<'a, 'tcx> {
-    fn check_rust_syntax(&self, item: &clean::Item, dox: &str, code_block: RustCodeBlock) {
+    fn check_rust_syntax(&self, _item: &clean::Item, dox: &str, code_block: RustCodeBlock) {
         let sess = ParseSess::new(FilePathMapping::empty());
         let source_file = sess.source_map().new_source_file(
             FileName::Custom(String::from("doctest")),
             dox[code_block.code].to_owned(),
         );
-
-        let errors = {
-            let mut lexer = Lexer::new(&sess, source_file, None);
-            while let Ok(token::Token { kind, .. }) = lexer.try_next_token() {
-                if kind == token::Eof {
-                    break;
-                }
-            }
-
-            let errors = lexer.buffer_fatal_errors();
```

Comment:
@GuillaumeGomez I think this chunk of code is mostly dead code today? Most lexing errors are emitted via `span_diagnostic` directly, and only a single special case (completely unknown token) goes via `buffer_fatal_errors`.

Reply:
I think too. Rustdoc needs some cleanup.